src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLSPOptions.cs

```diff
@@ -32,6 +32,13 @@ public static LogLevel GetLogLevelForTrace(Trace trace)
             _ => LogLevel.None,
         };
 
-    internal RazorLSPOptions With(ClientSettings clientSettings)
-        => new(Trace, EnableFormatting, AutoClosingTags, clientSettings);
+    /// <summary>
+    /// Initializes the LSP options with the settings from the passed in client settings, and default values for anything
+    /// not defined in client settings.
+    /// </summary>
+    internal static RazorLSPOptions From(ClientSettings clientSettings)
+        => new(Default.Trace,
+            Default.EnableFormatting,
+            clientSettings.AdvancedSettings.AutoClosingTags,
```

Comment:
This change meant making the "With" method override the VS Code AutoClosingTags setting with the VS AutoClosingTags setting. This is fine in the one place the method was called, but if anyone ever called it anywhere else, it could/would be bad. Also since _everything_ in the options code likes to return default settings instead of null, I couldn't make it reason about whether it should override or not.

So, cue laziness, and renaming the method means it now more clearly indicates that it will use the default values to VS Code settings, and overwrite with VS settings.